src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -86,61 +98,205 @@ public Bridge select(
+                                                                  List<Bridge> conferenceBridges,
+                                                                  String participantRegion)
     {
-        Bridge bridge = null;
+        Optional<Bridge> result = bridges.stream()
+            .filter(not(Bridge::isOverloaded))
+            .filter(selectFrom(conferenceBridges))
+            .filter(inRegion(participantRegion))
+            .findFirst();
 
-        // Prefer a bridge in the participant's region.
-        if (participantRegion != null)
+        if (result.isPresent())
         {
-            bridge = findFirstOperationalInRegion(bridges, participantRegion);
+            logger.info("bridge_selected,bridge=" + result.get() +
+                ",conference_bridges=" + conferenceBridges.stream().map(Bridge::toString).collect(Collectors.joining(", ")) +
+                "participant_region " + participantRegion);
+
+            totalNotLoadedAlreadyInConferenceInRegion++;
         }
 
-        // Otherwise, prefer a bridge in the local region.
-        if (bridge == null)
+        return result;
+    }
+
+    /**
+     * Finds the least loaded bridge in the participant's region that is not
+     * overloaded.
+     *
+     * @param bridges the list of operational bridges, ordered by load.
+     * @param participantRegion the set of bridges that are already used in the conference.
+     *
+     * @return an optional that contains a bridge that is not loaded and that is
+     * in the participant region.
+     */
+    private Optional<Bridge> notLoadedInRegion(List<Bridge> bridges,
+                                               String participantRegion)
+    {
+        Optional<Bridge> result = bridges.stream()
+            .filter(not(Bridge::isOverloaded))
+            .filter(inRegion(participantRegion))
+            .findFirst();
+
+        if (result.isPresent())
         {
-            bridge = findFirstOperationalInRegion(bridges, localRegion);
+            logger.info("bridge_selected,bridge=" + result.get() +
+                "participant_region " + participantRegion);
+
+            totalNotLoadedInRegion++;
         }
 
-        // Otherwise, just find the first operational bridge.
-        if (bridge == null)
+        return result;
+    }
+
+    /**
+     * Finds the least loaded conference bridge in the participant's region that
+     * is already handling the conference.
+     *
+     * @param bridges the list of operational bridges, ordered by load.
+     * @param conferenceBridges the set of bridges that are already used in the conference.
+     * @param participantRegion the participant region.
+     *
+     * @return an optional that contains the least loaded bridge that is already
+     * in the conference and that is in the participant region if it exists.
+     */
+    private Optional<Bridge> leastLoadedAlreadyInConferenceInRegion(List<Bridge> bridges,
```

Comment:
We don't need `bridges` here, and can just start with `conferenceBridges` since we filter to those anyway.

Reply:
The bridges list is prioritized by load whereas the conferenceBridges list is not. We should probably make this more obvious by changing the conferenceBridges argument to be a `Set`.